File: python/orderbook_stream.py
# Stream a live order book over WebSocket and print the top of book in place.
# Usage: python orderbook_stream.py [SYMBOL]   (Ctrl+C to stop)
import asyncio
import sys
import ccxt.pro as ccxtpro


async def main():
    symbol = sys.argv[1] if len(sys.argv) > 1 else 'BTC/USDT'
    exchange = ccxtpro.binance()
    try:
        while True:
            orderbook = await exchange.watch_order_book(symbol, 10)
            bid = orderbook['bids'][0] if orderbook['bids'] else (None, None)
            ask = orderbook['asks'][0] if orderbook['asks'] else (None, None)
            print(f"\r{symbol}  bid {bid[0]} ({bid[1]})  ask {ask[0]} ({ask[1]})   ", end='', flush=True)
    finally:
        await exchange.close()


try:
    asyncio.run(main())
except KeyboardInterrupt:
    pass

File: php/orderbook_stream.php
<?php
// Stream a live order book over WebSocket and print the top of book in place.
// Usage: php orderbook_stream.php [SYMBOL]   (Ctrl+C to stop)
require __DIR__ . '/vendor/autoload.php';

use React\Async;

$symbol = $argv[1] ?? 'BTC/USDT';

function stream($symbol) {
    return Async\async(function () use ($symbol) {
        $exchange = new \ccxt\pro\binance();
        while (true) {
            $orderbook = Async\await($exchange->watch_order_book($symbol, 10));
            $bid = $orderbook['bids'][0] ?? [null, null];
            $ask = $orderbook['asks'][0] ?? [null, null];
            printf("\r%s  bid %s (%s)  ask %s (%s)   ", $symbol, $bid[0], $bid[1], $ask[0], $ask[1]);
        }
    })();
}

Async\await(stream($symbol));
